soroban-env-host/src/budget.rs

```diff
@@ -98,12 +101,14 @@ impl CostType {
 
 /// We provide a "cost model" object that evaluates a linear expression:
 ///
-///    f(x) = a + b*x
+///    f(x) = N * (a + b * Option<x>)
```

Comment:
I .. am unsure about this. My instinct is that the iteration count shouldn't wind up in the cost model at all. I think iteration counts should be limited to the calibration harness code. Especially I am unsure about the thing you're doing here where you describe the iteration count as similar to the _batching_ count that we use in callers (but then in the actual batching calls, eg. in `MeteredClone`, you're not using this, but passing `1` for iteration and `Some(n_elts.saturating_mul(T::DECLARED_SIZE))`.

I think there are 3 separate issues at work here:

1. A statistical iteration count that runs the same trial multiple times to average-out noise during calibration.
2. Charging the budget for a "bulk operation" like memcpy or memcmp where there's both an "element size" and an "element count"
3. Making a single budget _charge call_ that does a "bulk" call to `Budget::charge` that combines multiple logically-separate calls to `Budget::charge`, to avoid too much CPU overhead from the budget-charging-machinery

Cases 2 and 3 look similar but they're not, right? Like a call to something like `memcpy` is case 2 and _not_ case 3, because a single call to `memcpy` has a fixed constant term `a` -- the overhead of calling the function and setting up the copy loop -- and its variable cost is entirely in the `b` term, not the `N` term. Case 3 only occurs when we have a charge that has an `N` term that varies, because you want to multiply `a*N` to capture "a batch of `N` calls to `Budget::charge` where each call has constant term `a`", right?

So .. looking through the code I am not sure case 3 ever actually happens (outside of calibration). Or at least that it's correct for it to happen. I see you're _not_ doing case 3 for things like `memcmp`, but _are_ doing it for things like `VecEntry`, which .. I guess is plausible? But then if that ever _is_ the case, you're no longer dealing with a simple linear model, you've got a _bi-linear_ model right? Which .. I don't think is actually what the model fitter is fitting, and also doesn't quite look like what the evaluation function is doing (also I suspect it's overkill for these few cases, i.e. I don't expect `N*(a+bx)` for `N` vector element accesses is going to be a lot different from `a+b(N*x)` which is just `a+bx` for a different `b` (which I think will get divided-out during measurement anyways).